### ISSUE TITLE
[yowsup v3.2.3] Cannot send a message - Exception: Sent keys were not accepted

### ISSUE THREAD

**Issue by leemgs** [tgalal/yowsup#2823]
**Describe the bug**

I could not send a message to a phone with the yowsup-cli command.

**Debug output**

```bash
rpi$ yowsup-cli demos  --config ~/.config/yowsup/82103902xxxx/config.json --debug  -s 82103071xxxx "Hello. This message is send from Rpi3/Rasbian OS Stretch."
D 2019-06-21 11:04:03,350 yowsup.config.manager - load_path(path=/home/pi/.config/yowsup/82103902xxx/config.json)
D 2019-06-21 11:04:03,351 yowsup.config.manager - Detected config type: json
D 2019-06-21 11:04:03,351 yowsup.config.manager - Opening config for reading
D 2019-06-21 11:04:03,352 yowsup.config.manager - Loading config
D 2019-06-21 11:04:03,353 yowsup.config.manager - Detecting config for username=82103902xxxx, dir=/home/pi/.config/yowsup/82103902xxxx/
D 2019-06-21 11:04:03,353 yowsup.config.manager - Trying /home/pi/.config/yowsup/82103902xxxx/config.json
D 2019-06-21 11:04:03,354 yowsup.config.manager - load_path(path=/home/pi/.config/yowsup/82103902xxxx/config.json)
D 2019-06-21 11:04:03,354 yowsup.config.manager - Detected config type: json
D 2019-06-21 11:04:03,355 yowsup.config.manager - Opening config for reading
D 2019-06-21 11:04:03,355 yowsup.config.manager - Loading config
yowsup-cli  v3.2.0
yowsup      v3.2.3

Copyright (c) 2012-2019 Tarek Galal
http://www.openwhatsapp.org

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://openwhatsapp.org/yowsup/donate


D 2019-06-21 11:04:03,893 yowsup.stacks.yowstack - Initializing stack
D 2019-06-21 11:04:03,894 yowsup.stacks.yowstack - Constructed Network Layer
D 2019-06-21 11:04:03,895 yowsup.stacks.yowstack - Constructed Noise Segments Layer
D 2019-06-21 11:04:03,900 yowsup.stacks.yowstack - Constructed Noise Layer
D 2019-06-21 11:04:03,901 yowsup.stacks.yowstack - Constructed Coder Layer
D 2019-06-21 11:04:03,902 yowsup.stacks.yowstack - Constructed Logger Layer
D 2019-06-21 11:04:03,903 yowsup.stacks.yowstack - Constructed <yowsup.layers.axolotl.layer_control.AxolotlControlLayer object at 0x75abecb0>
D 2019-06-21 11:04:03,985 yowsup.stacks.yowstack - Constructed Axolotl Layer - <yowsup.layers.axolotl.layer_receive.AxolotlReceivelayer object at 0x75ab1810>
D 2019-06-21 11:04:03,985 yowsup.stacks.yowstack - Constructed Authentication Layer - Messages Layer - Receipt Layer - Ack Layer - Presence Layer - Ib Layer - Iq Layer - notification Ib Layer - Iq Layer - Chatstate Layer - call Layer - Groups Iq Layer - Media Layer - Privacy Layer - Profiles Layer
D 2019-06-21 11:04:03,987 yowsup.stacks.yowstack - Constructed Interface Layer
D 2019-06-21 11:04:03,988 yowsup.layers.network.layer - Created asyncore dispatcher
I 2019-06-21 11:04:03,989 yowsup.layers.network.layer - Connecting to e2.whatsapp.net:443
D 2019-06-21 11:04:03,989 yowsup.layers.network.dispatcher.dispatcher_asyncore - connect(('e2.whatsapp.net', 443))
D 2019-06-21 11:04:04,212 yowsup.layers.network.dispatcher.dispatcher_asyncore - handle_connect
D 2019-06-21 11:04:04,212 yowsup.layers.network.layer - Connected
D 2019-06-21 11:04:04,213 yowsup.axolotl.factory - get_manager(username=82103902xxx)
D 2019-06-21 11:04:04,216 yowsup.axolotl.manager - Initialized AxolotlManager [username=82103902XXXX, db=/home/pi/.config/yowsup/82103902xxx/axolotl.db]
D 2019-06-21 11:04:04,217 yowsup.axolotl.manager - level_prekeys(force=False)
D 2019-06-21 11:04:04,345 yowsup.axolotl.manager - len(pending_prekeys) = 812
D 2019-06-21 11:04:04,349 yowsup.axolotl.manager - load_unsent_prekeys
I 2019-06-21 11:04:04,471 yowsup.axolotl.manager - Loaded 812 unsent prekeys
D 2019-06-21 11:04:04,472 yowsup.axolotl.factory - get_manager(username=82103902XXXX)
D 2019-06-21 11:04:04,476 yowsup.axolotl.manager - Initialized AxolotlManager [username=82103902XXXX, db=/home/pi/.config/yowsup/82103902XXXX/axolotl.db]
D 2019-06-21 11:04:04,476 yowsup.axolotl.factory - get_manager(username=82103902XXXX)
D 2019-06-21 11:04:04,480 yowsup.axolotl.manager - Initialized AxolotlManager [username=82103902XXXX, db=/home/pi/.config/yowsup/82103902XXXX/axolotl.db]
D 2019-06-21 11:04:04,481 yowsup.layers.noise.layer - Received auth event
D 2019-06-21 11:04:04,481 yowsup.config.manager - Detecting config for username=82103902XXXX, dir=/home/pi/.config/yowsup/82103902XXXX/
D 2019-06-21 11:04:04,482 yowsup.config.manager - Trying /home/pi/.config/yowsup/82103902XXXX/config.json
D 2019-06-21 11:04:04,482 yowsup.config.manager - load_path(path=/home/pi/.config/yowsup/82103902XXXX/config.json)
D 2019-06-21 11:04:04,483 yowsup.config.manager - Detected config type: json
D 2019-06-21 11:04:04,483 yowsup.config.manager - Opening config for reading
D 2019-06-21 11:04:04,484 yowsup.config.manager - Loading config
D 2019-06-21 11:04:04,485 yowsup.env.env - Env not set, setting it to android
D 2019-06-21 11:04:04,486 yowsup.env.env - Current env changed to android
D 2019-06-21 11:04:04,486 yowsup.layers.noise.layer - Performing handshake [username= 82103902XXXX, passive=True]
D 2019-06-21 11:04:04,487 yowsup.layers.noise.layer - Starting handshake worker
D 2019-06-21 11:04:05,059 yowsup.config.manager - Detecting config for username=82103902XXXX, dir=/home/pi/.config/yowsup/82103902XXXX/
D 2019-06-21 11:04:05,060 yowsup.config.manager - Trying /home/pi/.config/yowsup/82103902XXXX/config.json
D 2019-06-21 11:04:05,061 yowsup.config.manager - load_path(path=/home/pi/.config/yowsup/82103902XXXX/config.json)
D 2019-06-21 11:04:05,062 yowsup.config.manager - Detected config type: json
D 2019-06-21 11:04:05,062 yowsup.config.manager - Opening config for reading
D 2019-06-21 11:04:05,063 yowsup.config.manager - Loading config
D 2019-06-21 11:04:05,067 yowsup.common.tools - writePhoneData(phone=82103902XXXX, name=config.json, val=[omitted])
D 2019-06-21 11:04:05,068 yowsup.common.tools - Writing /home/pi/.config/yowsup/82103902XXXX/config.json
D 2019-06-21 11:04:05,073 yowsup.layers.logger.layer - rx:
<success location="vll" creation="1561123946" t="1561129444" props="25">
</success>

D 2019-06-21 11:04:05,075 yowsup.layers.protocol_iq.layer - starting ping thread.
D 2019-06-21 11:04:05,076 yowsup.layers.axolotl.layer_control - SHOULD FLUSH KEYS 812 NOW!!
D 2019-06-21 11:04:05,077 yowsup.axolotl.manager - load_latest_signed_prekey
D 2019-06-21 11:04:05,301 yowsup.layers.logger.layer - tx:
<iq to="s.whatsapp.net" xmlns="encrypt" type="set" id="1">
<list>
<key>
<id>
▒▒
HEX:fab7c8
</id>
<value>
▒Me▒▒fޚA▒▒▒q▒]▒▒M▒.ۅ▒TO▒▒ѧ▒T
HEX:844d65d7e666de9a41e6eeff71fba35df2fe4dcb2edb85ed544f929fd1a7a554
</value>
</key>
<key>
<id>
▒▒
HEX:fab7c9
</id>
<value>
▒▒յ▒▒▒>l▒QL▒▒X{v▒▒LO▒c'
HEX:a384d5b5121500f5cf02e63e6c1ca7514c19b9e605587b76c0f24c064ff26327
</value>
</key>
<key>
<id>




   . . . Omission . . . 
HEX:000000
</id>
<value>
▒▒oO▒▒0▒wM▒▒▒▒▒▒9▒6ݽ▒*ï^
HEX:e4d16f4fc8d030f5774d15ac92a11addcc17d83903a536ddbdea2ac3af01165e
</value>
<signature>
▒7▒)▒i▒Ͷ▒p▒ԋ=J^▒u
Ѳ▒q/M=▒▒?obU)▒"▒▒▒▒~▒▒]▒▒jB▒▒E▒r▒x
HEX:d00437831029d969e501cdb6f370add48b3d4a5e008e750e0cd1b2cb712f4d3da712d63f6f625529bb2294b0c4c97eb8cd5de0d56a421c8d9545e41872d9780e
</signature>
</skey>
</user>
</list>
</iq>

E 2019-06-21 10:59:49,484 yowsup.layers.network.dispatcher.dispatcher_asyncore - Traceback (most recent call last):
  File "/usr/lib/python2.7/asyncore.py", line 83, in read
    obj.handle_read_event()
  File "/usr/lib/python2.7/asyncore.py", line 449, in handle_read_event
    self.handle_read()
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/network/dispatcher/dispatcher_asyncore.py", line 48, in handle_read
    self.connectionCallbacks.onRecvData(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/network/layer.py", line 100, in onRecvData
    self.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/network/layer.py", line 103, in receive
    self.toUpper(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/noise/layer_noise_segments.py", line 39, in receive
    self.toUpper(bytes(data))
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/noise/layer.py", line 156, in receive
    self._flush_incoming_buffer()
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/noise/layer.py", line 144, in _flush_incoming_buffer
    self.toUpper(self._wa_noiseprotocol.receive())
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/coder/layer.py", line 21, in receive
    self.toUpper(node)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/logger/layer.py", line 14, in receive
    self.toUpper(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/axolotl/layer_control.py", line 35, in receive
    self.toUpper(protocolTreeNode)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 194, in receive
    s.receive(data)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/axolotl/layer_send.py", line 42, in receive
    if not self.processIqRegistry(protocolTreeNode):
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/__init__.py", line 161, in processIqRegistry
    successClbk(protocolTreeNode, originalIq)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/axolotl/layer_base.py", line 49, in onSuccess
    entity = ResultGetKeysIqProtocolEntity.fromProtocolTreeNode(resultNode)
  File "/usr/local/lib/python2.7/dist-packages/yowsup-3.2.3-py2.7.egg/yowsup/layers/axolotl/protocolentities/iq_keys_get_result.py", line 88, in fromProtocolTreeNode
    preKeyId = ResultGetKeysIqProtocolEntity._bytesToInt(preKeyNode.getChild("id").getData())
AttributeError: 'NoneType' object has no attribute 'getChild'

D 2019-06-21 10:59:49,485 yowsup.layers.network.dispatcher.dispatcher_asyncore - handle_close
D 2019-06-21 10:59:49,485 yowsup.layers.network.layer - Disconnected
D 2019-06-21 10:59:49,487 yowsup.layers.axolotl.layer_control - Disconnected, reboot_connect? = False
D 2019-06-21 10:59:49,488 yowsup.layers.protocol_iq.layer - stopping ping thread
xterm-256colorD 2019-06-21 10:59:49,831 yowsup.layers.protocol_iq.layer - YowPingThread-2 - ping thread stopped

```

**Config file**

```bash
$ cat ~/.config/yowsup/82103902xxxx/config.json
{
    "__version__": 1,
    "cc": "82",
    "client_static_keypair": "xxxxxxxxxxxxxxxxxxx7Ou6WOCltaPg5jhfUMFxDttL+f3rCPqFKwRKFIcedAA==",
    "edge_routing_info": "CAgIBQ==",
    "expid": "xxxxxxxxxxxx1OfaLAz2leg==",
    "fdid": "641d65d6-4ae3-4a5d-bb6a-d768b4430446",
    "id": "xxxxxxxxxhKdgN0u0pVE4=",
    "mcc": "450",
    "mnc": "10",
    "phone": "82103902xxxx",
    "server_static_public": "xxxxxxxxxxxxxxxxxxxpxpL8WVkwPZLMk3ybT3ggI=",
    "sim_mcc": "000",
    "sim_mnc": "000"

```

**Versions**
```
$ yowsup-cli version
yowsup-cli v3.2.0
Using yowsup v3.2.3
```

**To Reproduce**

* Build yowsup v3.2.3 from github.com
```
git clone git://github.com/tgalal/yowsup.git
cd yowsup
git checkout -b v3.2.3 v3.2.3
sudo python setup.py install
```

* Send a SMS message with yousup-cli command
```bash
yowsup-cli demos  --config ~/.config/yowsup/82103902XXXX/config.json --debug  -s 82103071xxxx "Hello. This message is send from Rpi3/Rasbian OS Stretch."

```
**Expected behavior**
Omission.

**OS (please complete the following information):**

- RPi3+
- Raspbian Stretch

**Additional context**
Omission.

**Comment by leemgs** [comment]
* Screenshot:
![image](https://user-images.githubusercontent.com/82404/59932047-4e7cbb00-9481-11e9-9dc6-5fa6352dfe39.png)

**Comment by leemgs** [comment]
* Related issues:
  * https://github.com/tgalal/yowsup/issues/2791

**Comment by leemgs** [comment]
Anyone that had already experienced this issue? Welcome to any hints.

**Comment by tgalal** [comment]
I'm not sure which part in the logs indicate a "Sent keys were not accepted", but in any case, the issue indicated in logs is already fixed by 4dc99cfc5d396b83b4777a37ea50526e4595830c but not yet released. I'll try to make a release during the weekend, in the mean time you try with code from master branch

**Comment by leemgs** [comment]
Thank you for your fast reply. 😄
